SimG4Core/Application/plugins/OscarMTProducer.cc

```diff
@@ -165,7 +177,13 @@ OscarMTProducer::OscarMTProducer(edm::ParameterSet const& p, const OscarMTMaster
   edm::LogVerbatim("SimG4CoreApplication") << "OscarMTProducer is constructed";
 }
 
-OscarMTProducer::~OscarMTProducer() {}
+OscarMTProducer::~OscarMTProducer() {
+  auto token = edm::ServiceRegistry::instance().presentToken();
```

Comment:
@Dr15Jones , is it safe calling edm::ServiceRegistry::instance() in a destructor? May be better keeping the pointer as a class member?

Reply:
The framework guarantees that it is safe to use the Service system in module destructors.

Reply:
+1

If the framework guarantee the order of destruction, it is fine.